tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_composite_conv.py

```diff
@@ -890,40 +863,41 @@ def conv_input_interleaved_to_sharded(self, conv_input_on_device: ttl.tensor.Ten
         padded_input_channels = conv_input_on_device.shape()[3]
         assert padded_input_channels >= input_channels
         assert padded_input_channels == self.padded_input_channels
-        act_c_num_blocks = self.opt_conv_block_conf_auto.act_c_num_blocks
         grid_size = (num_cores_w, num_cores_h)
 
         input_size_to_shard_evenly = _nearest_y(
             self.input_tensor_shape[0] * self.input_tensor_shape[1] * self.input_tensor_shape[2], num_cores_nhw * 32
         )
         untilize_with_halo_input_shard_height = (int)(input_size_to_shard_evenly / num_cores_nhw)
         # Convert interleaved to sharded
-        if act_c_num_blocks > 1:  # 2D conv
-            assert padded_input_channels % act_c_num_blocks == 0
-            assert (
-                not self.use_shallow_conv_variant
-            ), "Do not support shallow depth convs with 2d systolic variant. Run with use_1d_systolic_array=True or unset use_shallow_conv_variant. Default value of use_shallow_conv_variant is False."
+        if self.is_1d_systolic:
             conv_input_on_device = ttl.tensor.interleaved_to_sharded(
                 conv_input_on_device,
                 grid_size,
                 [
                     untilize_with_halo_input_shard_height,
-                    (int)(padded_input_channels / act_c_num_blocks),
+                    padded_input_channels,
                 ],  # act_block_w_datums may include reads of multiple pixels in window
-                ttl.tensor.TensorMemoryLayout.BLOCK_SHARDED,
-                ttl.tensor.ShardOrientation.COL_MAJOR,
+                ttl.tensor.TensorMemoryLayout.HEIGHT_SHARDED,
+                ttl.tensor.ShardOrientation.ROW_MAJOR,
             )
         else:
+            act_c_num_blocks = self.opt_conv_parall_conf_auto.grid_size.y
```

Comment:
IMO, this variable is clearer as just `grid_size_y` or something.  Or perhaps `grid_size_n` for matmul `mkn` dims.